content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,483 @@
+	}
+	log.Event(ctx, fmt.Sprintf("Read %d cdIDPairs" , len(cdIDPairs)), log.INFO)
+	ctx, err := authenticate(ctx, httpClient, environment, username, password)
+	if err != nil {
+		_ = fmt.Errorf("error occurred while authenticating. Stopping the script. error: %s", err.Error())
+		return
+	}
+
+	collectionID, err := createCollection(ctx, httpClient, getCollectionName(), environment)
+	if err != nil {
+		fmt.Errorf("error occurred while creating collection. Stopping the script. error: %s", err.Error())
+		return
+	}
+	log.Event(ctx, fmt.Sprintf("created collection: %s", collectionID), log.INFO)
+
+	for _, pair := range cdIDPairs {
+		pair.Print(ctx, "Processing ")
+
+		oldCdIDLocation, err := searchOldCdID(ctx, httpClient, environment, pair.oldCdID)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID location")
+			continue
+		}
+
+		existingCollectionID, err := checkIfCdIDExistsInAnotherCollection(ctx, httpClient, environment,oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while verifying if cdID location exists in another collection")
+			continue
+		}
+
+		if len(existingCollectionID) > 0 {
+			pair.Print(ctx, fmt.Sprintf("stopping. Error occurred. CDID: %s already exists in another collection: %s", pair.oldCdID, existingCollectionID))
+			continue
+		}
+
+		cdIDData, err := fetchDataForCDID(ctx, httpClient, environment, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID data.")
+			continue
+		}
+
+		err = addCDIDToCollection(ctx, httpClient, environment, collectionID, oldCdIDLocation, cdIDData)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while adding CDID to collection")
+			continue
+		}
+
+		err = approveCDID(ctx, httpClient, environment, collectionID, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while approving cdid in the collection")
+			continue
+		}
+
+		pair.Print(ctx, "Completed Processing ")
+	}
+
+	log.Event(ctx, "successfully updated all documents.", log.INFO)
+}
+
+func checkIfCdIDExistsInAnotherCollection(ctx context.Context, client *http.Client, environment string, cdIDLocation string) (string, error) {
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+	collectionCheckURL := fmt.Sprintf("%s/checkcollectionsforuri?uri=%s/data.json", environment, cdIDURI)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", collectionCheckURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to prepare search cdid in collection request. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		if err != nil {
+			errMessage := fmt.Errorf("failed to search in API for cdid in collection . Error: %v", err)
+			log.Error(errMessage)
+
+			return "", errMessage
+		}
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode > 400 {
+		return "existing-Collection-ID", nil
```

Comment:
I'm not sure I follow what this should be doing? I think this should be the collection name returned in the response if it exists?